huff_cli/src/huffc.rs

```diff
@@ -190,33 +216,72 @@ fn main() {
         cached: use_cache,
     };
 
-    if let Some(TestCommands::Test { format, match_ }) = cli.test {
-        match compiler.grab_contracts() {
-            Ok(contracts) => {
-                let match_ = Rc::new(match_);
-
-                for contract in &contracts {
-                    let tester = HuffTester::new(contract, Rc::clone(&match_));
-
-                    let start = Instant::now();
-                    match tester.execute() {
-                        Ok(res) => {
-                            print_test_report(res, ReportKind::from(&format), start);
+    if let Some(sub) = cli.sub {
```

Comment:
move all this complexity in other functions, with better error handling using eyre ?